software/controller/test/uart/uart_test.cpp

```diff
@@ -47,20 +47,25 @@ TEST(UartTest, Tx) {
   // create a buffer slightly bigger than the tx buffer
   size_t buffer_size{test_uart.GetBufferLength() + 1};
   uint8_t send_buffer[buffer_size];
-  for (uint8_t i = 0; i < buffer_size; ++i) send_buffer[i] = i;
+  for (size_t i = 0; i < buffer_size; ++i) send_buffer[i] = static_cast<uint8_t>(i);
 
   // reasonably sized write
   size_t bytes_added{50};
   ASSERT_EQ(test_uart.Write(send_buffer, bytes_added), bytes_added);
+  ASSERT_TRUE(test_uart.TxInProgress());
   ASSERT_EQ(test_uart.TxFree(), test_uart.GetBufferLength() - bytes_added);
   for (uint8_t i = 0; i < bytes_added; ++i) ASSERT_EQ(*test_uart.GetTxByte(), i);
 
+  // we gathered all bytes from the Tx buffer, Tx is no longer in progress
+  ASSERT_FALSE(test_uart.TxInProgress());
+
   // buffer should be empty again
   ASSERT_EQ(test_uart.TxFree(), test_uart.GetBufferLength());
   // fill buffer through long write
   ASSERT_EQ(test_uart.Write(send_buffer, buffer_size), test_uart.GetBufferLength());
   // check that we cannot write to buffer anymore
   ASSERT_EQ(test_uart.Write(send_buffer, 1), 0);
   // check contents of buffer
-  for (uint8_t i = 0; i < test_uart.GetBufferLength(); ++i) ASSERT_EQ(*test_uart.GetTxByte(), i);
+  for (size_t i = 0; i < test_uart.GetBufferLength(); ++i)
```

Comment:
This is actually a good example of why omitting {} are bad. Here it is actually dangerous because
unless you go and check you cannot assume that ASSERT() is't a multi-line macro (even if it probably isn't)